src/main/java/com/extremelyd1/gameboard/GameBoard.java

```diff
@@ -61,7 +61,19 @@ public GameBoard(Game game) {
     public void update() {
         // Clear all existing scores
         for (String entry : this.scoreboard.getEntries()) {
-            this.scoreboard.resetScores(entry);
+            Boolean entryStillExists = false;
+
+            for (int i = 0; i < boardEntries.size(); i++) {
```

Comment:
There is no reason to use an indexed for loop here, `boardEntries` is a `List` and implements `Iterable`.

Reply:
changed